### PR TITLE
Initial security policy

### LINKED ISSUE #1617: Update Security Policy instead of Referring to CoreOS Security Policy
The README links to the outdated CoreOS security policy.  It would be great if dexidp could write it's own security policy and list it both in the readme and the new security policy part of github.  Thanks!

### LINKED ISSUE #1867: Add SECURITY.md
Dex needs a security disclosure process :)

### PR DESCRIPTION
**Pull request by sagikazarmark**
This is an initial version of the security policy. I've already requested a dedicated email address (`security@dexidp.io`) for this purpose. I will update the policy once that's set up, but it shouldn't block this PR.

Fixes #1617
Fixes #1867

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -102,13 +102,9 @@ All changes or deprecations of connector features will be announced in the [rele
 * Client libraries
   * [Go][go-oidc]
 
-## Reporting a security vulnerability
+## Reporting a vulnerability
```
Comment (nabokihms) [inline README.md:105]: We also need to update [the index page](https://github.com/dexidp/website/blob/main/content/docs/_index.md#reporting-a-security-vulnerability) on the website.

File: .github/SECURITY.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# Security Policy
+
+## Reporting a vulnerability
+
+To report a vulnerability, send an email to [cncf-dex-maintainers@lists.cncf.io](mailto:cncf-dex-maintainers@lists.cncf.io)
+detailing the issue and steps to reproduce. The reporter(s) can expect a
+response within 24 hours acknowledging the issue was received. If a response is
```
Comment (nabokihms) [inline .github/SECURITY.md:7]: Maybe it is better to take 2 working days for the first response? Vacations and emergencies can slow down the process. I'm just trying to play it safe :)

**Review by nabokihms — APPROVED** [review]
Looks good enough, thanks! I left a few comments, though nothing serious.

**Review by justaugustus — APPROVED** [review]
We may end up tweaking some of the language, but I think this is a great update!
Thanks @sagikazarmark!

**Pull request merged**